Feature/#104 리뷰작성화면 화면이동 및 서버통신 로직 제작

sero3779 opened the pull request:
안녕하세요 클로이~ @anyukyung 
리뷰 작성화면에 필요한 화면이동 로직 및 서버통신 로직이 제작되었습니다~

## 🧴 PR 요약
- 팝업 화면에 표시할 정보 제작
- 팝업화면의 자세히보기 눌렀을때의 동작 제작
- 팝업화면 이동 분기처리 로직 (레벨업 여부 체크) 제작
- 유저 레벨 화면 변경사항 반영, 뷰모델 제작

#### 📌 변경 사항
- UserLevelInfo가 자체적으로 남은 리뷰수를 들고있을 필요가 없기에 제거하였습니다.
- 레벨 관련 이미지 에셋이 추가되었습니다.
- 다음 레벨을 반환하는 변수와 다음레벨까지 남은 리뷰수를 알려주는 변수를 제작하였습니다.

##### 📸 ScreenShot

https://user-images.githubusercontent.com/67148595/216304601-cf8ef9bf-df1d-4e26-84ae-ce000ef28a30.mp4


#### Linked Issue
close #104

sero3779 committed:
[Feat] #104 마이페이지, 닉네임 화면 의존성 주입,  coordinator 연결

sero3779 committed:
[Fix] #104 page control 색상 잘못 들어있던 것 수정

sero3779 committed:
Merge branch 'develop' into feature/#104

sero3779 committed:
[Feat] #104 리뷰작성화면에서 탭바 숨기기

sero3779 committed:
[Feat] #104 리뷰 등록 위해 StoreId 인자로 받아오도록 변경

sero3779 committed:
[Feat] #104 Register Review 버튼 탭시 동작 구현

sero3779 committed:
[Feat] #104 리뷰 등록이후 팝업 화면 이동 로직 제작

sero3779 committed:
[Chore] #104 레벨 이미지 추가

sero3779 committed:
[Feat] #104 LevelInfo에 남은리뷰수 삭제, Level에 다음레벨 정보 추가

sero3779 committed:
[Feat] #104 리뷰 가져오기 로직 제작, 레벨업 여부 확인 로직 제작

sero3779 committed:
[Feat]  #104 가져온 리뷰수 기준으로 화면 이동 분기처리 로직 제작

sero3779 committed:
[Feat] #104 중복해서 리뷰업로드 되는 것 방지 위해 로직 추가

sero3779 committed:
[Feat] #104 자세히 알아보기 탭했을때 동작 구현

sero3779 committed:
[Fix] MockTask self 참조카운트 없어서 nil되던 것 수정

sero3779 committed:
[Feat] #104 유저 레벨 화면 이동 로직 제작

sero3779 committed:
[Feat] #104 회원 등급 안내 화면 변경사항 반영

sero3779 committed:
[Feat] #104 UserLevelViewmodel 제작

sero3779 committed:
[Feat] #104 ViewModel 통해 받아온 데이터 반영하는 로직 제작

anyukyung commented on RefillStation/RefillStation/Application/Coordinator/RegisterReviewCoordinator.swift:
```diff
@@ -21,4 +21,26 @@ final class RegisterReviewCoordinator: Coordinator {
         navigationController.pushViewController(registerReviewViewController, animated: true)
     }
+
+    func registerReviewSuccessed(userLevel: UserLevelInfo.Level?) {
```
Succeed 로 표현하는게 좀 더 적절하지 않을까 합니다!

sero3779 replied:
succeeded가 문법적으로 맞는 표현이라 이렇게 수정하겠습니다~!

anyukyung commented on RefillStation/RefillStation/Domain/Entities/UserLevelInfo.swift:
```diff
@@ -37,6 +37,30 @@ struct UserLevelInfo: Hashable {
+            case .beginner:
+                return 2
+            case .prospect:
+                return 2
```
추후에 변경되지 않는다면 이렇게 처리해도 될 듯 하네요 !!
```swift
            case .beginner, .prospect:
                return 2
```

anyukyung commented on RefillStation/RefillStation/Domain/Entities/UserLevelInfo.swift:
```diff
@@ -37,6 +37,30 @@ struct UserLevelInfo: Hashable {
         }
+        var nextLevel: Level {
+            switch self {
+            case .regular:
+                return .beginner
+            case .beginner:
+                return .prospect
+            case .prospect:
+                return .fancier
+            case .fancier:
+                return .fancier
```
number Type 열거형으로 선언해서 다음 레벨을 표현하는 방법도 좋을 것 같아요!

anyukyung commented on RefillStation/RefillStation/Presentation/RegisterReviewScene/ViewController/RegisterReviewViewController.swift:
```diff
@@ -25,10 +25,14 @@ final class RegisterReviewViewController: UIViewController {
         button.isEnabled = false
+        button.addAction(UIAction { _ in
+            self.viewModel.registerButtonTapped()
+            button.isEnabled = false
```
이부분은 여러번 클릭하는 걸 막기 위해서 false로 바꾸신 거 맞을까요!?

sero3779 replied:
맞습니다~!

sero3779 committed:
[Refactor] #104 스위치문 간략화, 네이밍 개선

- beginner와 prospect의 다음레벨까지 써야하는 리뷰개수가 같으므로 이를 통합
- Successed라는 네이밍을 문법에 맞게 Succeeded로 변경
